rules-tests/DowngradePhp70/Rector/Instanceof_/DowngradeInstanceofThrowableRector/Fixture/already-processed.php.inc

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Rector\Tests\DowngradePhp70\Rector\Instanceof_\DowngradeInstanceofThrowableRector\Fixture;
+
+use Throwable;
+
+return something() && (($throwable = $e) instanceof Throwable || $throwable instanceof \Exception);
+
+?>
```

Comment:
could you add 1 more for $throwable instanceof Exception in the left and instanceof Throwable in the right?

Reply:
That will currently be processed into `$throwable instanceof \Exception || (($throwable = $e) instanceof Throwable || $throwable instanceof \Exception)` . Should I handle that case as well? Because there is infinite number of isomorphisms that we could handle as well.

Reply:
That's why BooleanOr need to be added to node types so you can look up conditions via BetterNodeFinder, or by BinaryOpConditonsCollector

https://github.com/rectorphp/rector-src/blob/34c61ba4498150972db843fca807416540ca71e8/packages/NodeCollector/BinaryOpConditionsCollector.php#L10

Loop the condition, found instanceof, run refactor process

Reply:
Now it uses `BinaryOpConditonsCollector` so it supports arbitrary left-associative tree.